src/Compilers/CSharp/Portable/Binder/Semantics/Operators/UnaryOperatorSignature.cs

```diff
@@ -25,7 +25,27 @@ public UnaryOperatorSignature(UnaryOperatorKind kind, TypeSymbol operandType, Ty
 
         public override string ToString()
         {
-            return $"kind: {this.Kind} operand: {this.OperandType} return: {this.ReturnType}";
+            return $"kind: {this.Kind} operandType: {this.OperandType} operandRefKind: {this.RefKind} return: {this.ReturnType}";
+        }
+
+        public RefKind RefKind
+        {
+            get
+            {
+                if (Method != null)
+                {
+                    Debug.Assert(Method.ParameterCount == 1);
+
+                    if (!Method.ParameterRefKinds.IsDefaultOrEmpty)
+                    {
+                        Debug.Assert(Method.ParameterRefKinds.Length == 1);
+
+                        return Method.ParameterRefKinds.Single();
```

Comment:
>.Single() [](start = 55, length = 9)

Is ```Single()``` better than just ```[0]```?

Reply:
It will throw a more descriptive exception if something went wrong.

---
In reply to: [155394341](https://github.com/dotnet/roslyn/pull/23122#discussion_r155394341) [](ancestors = 155394341)